desc/objectives/_coils.py

```diff
@@ -1599,3 +1612,211 @@ def compute(self, params, constants=None):
+        ), (
+            "surface_current_field must be a CurrentPotentialField or "
+            + f"FourierCurrentPotentialField, instead got {type(surface_current_field)}"
+        )
+        self._surface_current_field = surface_current_field
+        self._source_grid = source_grid
+
+        super().__init__(
+            things=[surface_current_field],
+            target=target,
+            bounds=bounds,
+            weight=weight,
+            normalize=normalize,
+            normalize_target=normalize_target,
+            loss_function=loss_function,
+            deriv_mode=deriv_mode,
+            name=name,
+        )
+
+    def build(self, use_jit=True, verbose=1):
+        """Build constant arrays.
+
+        Parameters
+        ----------
+        use_jit : bool, optional
+            Whether to just-in-time compile the objective and derivatives.
+        verbose : int, optional
+            Level of output.
+
+        """
+        from desc.magnetic_fields import FourierCurrentPotentialField
+
+        surface_current_field = self.things[0]
+        if isinstance(surface_current_field, FourierCurrentPotentialField):
+            M_Phi = surface_current_field._M_Phi
+            N_Phi = surface_current_field._N_Phi
+        else:
+            M_Phi = surface_current_field.M + 1
+            N_Phi = surface_current_field.N + 1
+
+        if self._source_grid is None:
+            source_grid = LinearGrid(
+                M=3 * M_Phi + 1,
+                N=3 * N_Phi + 1,
+                NFP=surface_current_field.NFP,
+            )
+        else:
+            source_grid = self._source_grid
+
+        if not np.allclose(source_grid.nodes[:, 0], 1):
+            warnings.warn("Source grid includes off-surface pts, should be rho=1")
+
+        # source_grid.num_nodes for the regularization cost
+        self._dim_f = source_grid.num_nodes
+        self._surface_data_keys = ["K", "|e_theta x e_zeta|"]
+
+        timer = Timer()
+        if verbose > 0:
+            print("Precomputing transforms")
+        timer.start("Precomputing transforms")
+
+        surface_transforms = get_transforms(
+            self._surface_data_keys,
+            obj=surface_current_field,
+            grid=source_grid,
+            has_axis=source_grid.axis.size,
+        )
+        if self._normalize:
+            if isinstance(surface_current_field, FourierCurrentPotentialField):
+                self._normalization = np.max(
+                    [abs(surface_current_field.I) + abs(surface_current_field.G), 1]
+                )
+            else:  # it does not have I,G bc is CurrentPotentialField
+                Phi = surface_current_field.compute("Phi", grid=source_grid)["Phi"]
+                self._normalization = np.max(
+                    [
+                        np.mean(np.abs(Phi)),
+                        1,
+                    ]
+                )
```

Comment:
Is using `I` and `G` a better normalization factor if those attributes exist? Would it be better/more consistent to always use `Phi` for the normalization for both classes?

Reply:
not necessarily, I guess we could just always compute Phi. But if I and G exist and are not zero, they set the scale pretty well (as usually the field initially has zero for Phi_mn)

Reply:
It would be nice to always use `Phi` so the normalization is consistent, but I realize the most common use case is a `FourierCurrentPotentialField` with `Phi=0` so I guess this is fine